dev-python/QtPy/QtPy-1.9.0-r3.ebuild

```diff
@@ -0,0 +1,84 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..8} )
+
+inherit distutils-r1 virtualx
+
+DESCRIPTION="Abstraction layer on top of PyQt5 and PySide2 and additional custom QWidgets"
+HOMEPAGE="https://github.com/spyder-ide/qtpy"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+
+IUSE="
+	3d bluetooth charts concurrent datavis dbus declarative designer
+	examples gles2-only gui help location multimedia network networkauth
+	opengl positioning printsupport qml quick script scripttools
+	scxml sensors serialport speech sql ssl svg test testlib webchannel
+	webengine webkit websockets widgets x11extras xml xmlpatterns
+"
+
+RDEPEND="
+	app-eselect/eselect-QtPy
+	|| (
+		dev-python/PyQt5[${PYTHON_USEDEP},designer?,gles2-only?,gui?,help?,location?,multimedia?,network?,opengl?,positioning?,printsupport?,sensors?,serialport?,sql?,svg?,testlib?,webchannel?,websockets?,widgets?,x11extras?,xml(+)?,xmlpatterns?]
+		dev-python/pyside2[${PYTHON_USEDEP},designer?,gles2-only?,gui?,help?,location?,multimedia?,network?,opengl(+)?,positioning?,printsupport?,sensors?,serialport(+)?,sql?,svg?,testlib?,webchannel?,websockets?,widgets?,x11extras?,xml?,xmlpatterns?]
+	)
+
+	webengine? ( || (
+		dev-python/PyQtWebEngine[${PYTHON_USEDEP}]
+		dev-python/pyside2[${PYTHON_USEDEP},webengine]
+	) )
+
+	qml? ( || (
+		dev-python/PyQt5[${PYTHON_USEDEP},declarative]
+		dev-python/pyside2[${PYTHON_USEDEP},qml]
+	) )
+
+	quick? ( || (
+		dev-python/PyQt5[${PYTHON_USEDEP},declarative]
+		dev-python/pyside2[${PYTHON_USEDEP},quick]
+	) )
+
+	declarative? ( || (
+		dev-python/PyQt5[${PYTHON_USEDEP},declarative]
+		dev-python/pyside2[${PYTHON_USEDEP},qml,quick]
+	) )
+
+	bluetooth? ( dev-python/PyQt5[${PYTHON_USEDEP},bluetooth] )
+	dbus? ( dev-python/PyQt5[${PYTHON_USEDEP},dbus] )
+	examples? ( dev-python/PyQt5[${PYTHON_USEDEP},examples] )
+	networkauth? ( dev-python/PyQt5[${PYTHON_USEDEP},networkauth] )
+	ssl? ( dev-python/PyQt5[${PYTHON_USEDEP},ssl] )
+	webkit? ( dev-python/PyQt5[${PYTHON_USEDEP},webkit] )
+
+	3d? ( dev-python/pyside2[${PYTHON_USEDEP},3d] )
+	charts? ( dev-python/pyside2[${PYTHON_USEDEP},charts] )
+	concurrent? ( dev-python/pyside2[${PYTHON_USEDEP},concurrent] )
+	datavis? ( dev-python/pyside2[${PYTHON_USEDEP},datavis] )
+	scxml? ( dev-python/pyside2[${PYTHON_USEDEP},scxml] )
+	script? ( dev-python/pyside2[${PYTHON_USEDEP},script] )
+	scripttools? ( dev-python/pyside2[${PYTHON_USEDEP},scripttools] )
+	speech? ( dev-python/pyside2[${PYTHON_USEDEP},speech] )
```

Comment:
But these make no sense to me.

Reply:
The thing is that PyQt5 and PySide2 kinda do the same thing, but they are not identical. And some Python bindings for Qt modules are only provided by the one or the other. So for example, if I want to have the Python bindings for dev-qt/qtbluetooth I have to install PyQt5, PySide2 does not provided those bindings. If on the other hand I want to have the Python bindings for dev-qt/qt3d I have to install PySide2, PyQt5 does not provide those bindings. If I want to have bindings for both dev-qt/qtbluetooth *and* dev-qt/qt3d, than I have to install both. Does this clear up why I did what I did here?

Reply:
Hmm I see, an alternative would be to control the deps directly via the deps which might not be clear to our users. Let me think about it for a while, and ask someone else to check.

To me this looks like a timebomb waiting to be exploded :)